src/common/token_bucket.h

```diff
@@ -11,15 +11,64 @@
 
 #include "torint.h"
 
-typedef struct token_bucket_rw_t {
+/** Largest allowable burst value for a token buffer. */
+#define TOKEN_BUCKET_MAX_BURST INT32_MAX
+
+/** A generic token buffer configuration: determines the number of tokens
+ * added to the bucket in each time unit (the "rate"), and the maximum number
+ * of tokens in the bucket (the "burst") */
+typedef struct token_bucket_cfg_t {
   uint32_t rate;
   int32_t burst;
-  int32_t read_bucket;
-  int32_t write_bucket;
-  uint32_t last_refilled_at_ts;
-} token_bucket_rw_t;
+} token_bucket_cfg_t;
+
+/** A raw token bucket, decoupled from its configuration and timestamp. */
+typedef struct token_bucket_raw_t {
+  int32_t bucket;
+} token_bucket_raw_t;
+
+/** A timestamp for a token bucket. The units of this timestamp are
+ * unspecified, but must match with the rate set in the token_bucket_cfg_t. */
+typedef struct token_bucket_timestamp_t {
+  uint32_t last_refilled_at;
```

Comment:
If this is a Unix timestamp, this means we are stuck in 2038 ... :S  Could we go to uint64_t to avoid that?